src/posts/posts-dao.ts

```diff
@@ -26,6 +26,19 @@ export const getPostById = async (postId: string) => {
   .populate("groupId")
 }
 
+// Get posts by recipeId
+export const getPostsByRecipeId = async (recipeId: string) => {
+  const posts = await postsModel
+    .find({ recipeId: recipeId })
+    .populate("recipeId")
+    .populate("userId")
+    .populate("groupId")
+    .sort({ date: -1 });
+
+  const texts = posts.map((post: any) => post.text);
```

Comment:
For extensibility, I would just return the whole post since that doesn't cost extra. but this is fine if it works